src/common/I18n.ts

```diff
@@ -1,14 +1,22 @@
+type Translations = Record<string, Record<string, string>>;
+
+/**
+ * I18n translation class.
+ */
 export class I18n {
+  /** The language locale. */
   locale: string | undefined;
```

Comment:
How about change this line to locale?: string;
I would like to unify to your convention in src/common/utils.ts
```
secondaryMessage?: string;
```